src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -375,125 +379,128 @@ private static final class EnumMap {
-                                cb.checkcast(Number.class.describeConstable().get());
-                                cb.invokevirtual(Number.class.describeConstable().get(),
-                                                 "intValue",
-                                                 MethodType.methodType(int.class)
-                                                           .describeConstable()
-                                                           .get());
-                                cb.goto_(compare);
-                                cb.labelBinding(notNumber);
-                                cb.aload(0);
-                                cb.instanceof_(Character.class.describeConstable().get());
-                                cb.ifeq(next);
-                                cb.aload(0);
-                                cb.checkcast(Character.class.describeConstable().get());
-                                cb.invokevirtual(Character.class.describeConstable().get(),
-                                                 "charValue",
-                                                 MethodType.methodType(char.class)
-                                                           .describeConstable()
-                                                           .get());
-                                cb.labelBinding(compare);
-                                cb.ldc(integerLabel);
-                                cb.if_icmpne(next);
-                            } else {
-                                throw new InternalError("Unsupported label type: " + element.label().getClass());
-                            }
-                            cb.constantInstruction(idx);
-                            cb.ireturn();
+                        lastLabel = currentLabel;
+                        cases.add(new Element(target, next, currentLabel));
+                        switchCases.add(SwitchCase.of(idx, target));
+                    }
+                    cases = cases.reversed();
+                    switchCases = switchCases.reversed();
+                    cb.tableswitch(0, labels.length - 1, dflt, switchCases);
+                    for (int idx = 0; idx < cases.size(); idx++) {
+                        Element element = cases.get(idx);
+                        Label next = element.next();
+                        cb.labelBinding(element.target());
+                        if (element.caseLabel() instanceof Class<?> classLabel &&
+                            classLabel.describeConstable().isPresent()) {
+                            cb.aload(0);
+                            cb.instanceof_(classLabel.describeConstable().orElseThrow());
+                            cb.ifeq(next);
+                        } else if (element.caseLabel() instanceof Class<?> classLabel) {
+                            cb.aload(3);
+                            cb.constantInstruction(extraClassLabels.size());
+                            cb.aaload();
```

Comment:
Arrays are mutable in Java, so the VM can not know if the array of non denotable classes (`extraClassLabels`) will be changed or not so the result of aaload is not a constant so the call to isInstance can not be optimized. Using a immutable list (`List.of()`) instead of an array should work, because all the implementation of List.of() are using @Stable. In that case aaload becomes invokevirtual List.get().

Reply:
Done, thanks.